features/attach_invalidtoken.feature

```diff
@@ -21,3 +21,22 @@ Feature: Command behaviour when trying to attach a machine to an Ubuntu
            | focal   |
            | groovy  |
            | hirsute |
+
+    @uses.config.contract_token_staging_expired
+    @series.all
+    Scenario Outline: Attach command failure on expired token
+       Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I attempt to attach `contract_token_staging_expired` with sudo
+        Then stderr matches regexp:
+             """
+             Attach denied:
+             Contract ".*" .*
```

Comment:
Do we need make the Contract message here this permissive or can we add a more specific message here ?

Reply:
@lucasmoura Right now the test is written to accept any of the three scenarios (expired, not-yet-effective, never-effective), so that the exact status of the token in `UACLIENT_BEHAVE_CONTRACT_TOKEN_STAGING_EXPIRED` is more flexible. Currently, I think the one I put in jenkins has the status `never-effective`.

I'm open to making the test more specific to one of the three scenarios - we may have to ask the contract team for a custom token if we decide to test the expired or not-yet-effective scenarios. What do you think?